frontend/packages/policy-editor/src/components/PolicyCardRules/PolicyRule/PolicyAccessPackages/PolicyAccessPackages.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { type ReactElement, useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import {
+  StudioAlert,
+  StudioLabelAsParagraph,
+  StudioParagraph,
+  StudioTextfield,
+} from '@studio/components';
+import { getUpdatedRules } from '../../../../utils/PolicyRuleUtils';
+import { usePolicyEditorContext } from '../../../../contexts/PolicyEditorContext';
+import { usePolicyRuleContext } from '../../../../contexts/PolicyRuleContext';
+import classes from './PolicyAccessPackages.module.css';
+import { PolicyAccessPackageAccordion } from './PolicyAccessPackageAccordion';
+import { PolicyAccordion } from './PolicyAccordion';
+import {
+  filterAccessPackagesBySearchString,
+  groupAccessPackagesByArea,
+} from './policyAccessPackageUtils';
+import type { PolicyAccessPackageArea } from '@altinn/policy-editor';
+
+export const PolicyAccessPackages = (): ReactElement => {
+  const { t } = useTranslation();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const { policyRules, accessPackages, setPolicyRules, savePolicy } = usePolicyEditorContext();
+  const { policyRule } = usePolicyRuleContext();
+
+  const [chosenAccessPackages, setChosenAccessPackages] = useState<string[]>(
+    policyRule.accessPackages,
+  );
+
+  const groupedAccessPackagesByArea = useMemo(() => {
+    return groupAccessPackagesByArea(accessPackages);
+  }, [accessPackages]);
+
+  const handleSelectAccessPackage = (packageUrn: string): void => {
+    const isChecked = chosenAccessPackages.includes(packageUrn);
+    if (isChecked) {
+      setChosenAccessPackages((oldUrns) => oldUrns.filter((urn) => urn !== packageUrn));
+      const urnsToSave = policyRule.accessPackages.filter((x) => x !== packageUrn);
+      handleAccessPackageChange(urnsToSave);
+    } else {
+      setChosenAccessPackages((oldUrns) => [...oldUrns, packageUrn]);
+      const urnsToSave = [...policyRule.accessPackages, packageUrn];
+      handleAccessPackageChange(urnsToSave);
+    }
+  };
```

Comment:
Can we refactor this to increase readability? 😄 
It seems like there is similar code on line 38-40 and 42-44, maybe we can create a utils function for it? 
Anyways, I believe it will increase readability to put the content inside `if` and `else` in to separate functions that do one thing only 😄